litert/vendors/intel_openvino/dispatch/dispatch_api.cc

```diff
@@ -199,27 +219,37 @@ LiteRtStatus DispatchInvocationContextDestroy(
 LiteRtStatus DispatchAttachInput(
     LiteRtDispatchInvocationContext invocation_context, int graph_input_index,
     LiteRtTensorBufferHandle tensor_buffer_handle) {
-  if (auto status = invocation_context->AttachInput(graph_input_index,
-                                                    tensor_buffer_handle);
-      !status) {
-    LITERT_LOG(LITERT_ERROR, "Failed to attach input: %s",
-               status.Error().Message().c_str());
-    return status.Error().Status();
+  try {
+    if (auto status = invocation_context->AttachInput(graph_input_index,
+                                                      tensor_buffer_handle);
+        !status) {
+      LITERT_LOG(LITERT_ERROR, "Failed to attach input: %s",
+                status.Error().Message().c_str());
+      return status.Error().Status();
+    }
+    return kLiteRtStatusOk;
+  } catch (const std::exception& e) {
+    LITERT_LOG(LITERT_ERROR, "Exception in Dispatch attach_input: %s", e.what());
+    return kLiteRtStatusErrorRuntimeFailure;
   }
-  return kLiteRtStatusOk;
 }
 
 LiteRtStatus DispatchAttachOutput(
     LiteRtDispatchInvocationContext invocation_context, int graph_output_index,
     LiteRtTensorBufferHandle tensor_buffer_handle) {
-  if (auto status = invocation_context->AttachOutput(graph_output_index,
-                                                     tensor_buffer_handle);
-      !status) {
-    LITERT_LOG(LITERT_ERROR, "Failed to attach output: %s",
-               status.Error().Message().c_str());
-    return status.Error().Status();
+  try {
+    if (auto status = invocation_context->AttachOutput(graph_output_index,
+                                                      tensor_buffer_handle);
+        !status) {
+      LITERT_LOG(LITERT_ERROR, "Failed to attach output: %s",
+                status.Error().Message().c_str());
+      return status.Error().Status();
+    }
+    return kLiteRtStatusOk;
+  } catch (const std::exception& e) {
+    LITERT_LOG(LITERT_ERROR, "Exception in Dispatch attach_output: %s", e.what());
+    return kLiteRtStatusErrorRuntimeFailure;
```

Comment:
Thanks, this is exactly the pattern I had in mind!

I see that you haven't applied this to all function though, e.g. `DispatchDetachInput`. Presumably because you know that the underlying code doesn't throw? Though I am wondering if it would be best to simply wrap "all code", mainly because:

- If the underlying openvino impl changes or someone later changes `DispatchDetachInput` to invoke a function (perhaps indirectly) that might throw, then we'll be in trouble.
- If all functions are ``try/catch`-guarded then the pattern is pretty clear to anyone who makes changes (and perhaps adds new functions).

Reply:
I added exception handling only at required places as it might introduce additional overheads and also clutters the code.

The exception is currently added around all codes that has any calls into the OpenVINO headers/libs.
We'll introduce the exception handlings whenever any new code calls into the Openvino?